ttnn/cpp/ttnn/operations/conv/conv2d/conv2d_utils.cpp

```diff
@@ -802,38 +814,78 @@ ttnn::operations::matmul::MatmulProgramConfig determine_matmul_op_config_from_co
     }
 }
 
-void adjust_conv_op_config_for_auto_shard_if_necessary(
+std::tuple<uint32_t, uint32_t> get_padded_subblock_h_ntiles(
```

Comment:
you should use this f() in ttnn/cpp/ttnn/operations/conv/conv2d/device/conv2d_op_sharded_program_factory.cpp
and remove redundant code